src/main/java/com/codeofus/rent_a_park/models/Spot.java

```diff
@@ -21,15 +22,21 @@ public class Spot implements Serializable {
     Integer id;
 
     String address;
-    LocalDateTime availability;
-    Integer capacity;
+
+    String zone;
 
     @ManyToOne
     @JoinColumn(name = "renter_id", nullable = false)
     Person renter;
 
-    @ManyToOne
-    @JoinColumn(name = "parker_id")
-    Person parker;
+    @OneToMany(mappedBy = "spot", cascade = CascadeType.ALL, orphanRemoval = true)
+    List<Reservation> reservation = new ArrayList<>();
```

Comment:
Think about how would you model the data if you had to save it into the Mongo as one big json... what would be root and what would be it's children... apply same to the model to eliminate bidirectional connections + possible unnecessary entities (only the roots of your imagined MongoDB should be accessed via the repository)